src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double SCALE_FACTOR = 1.5;
+    private static final int FIRST_INDEX = 0;
+    private int size;
+    private Object[] arrList;
+
+    public ArrayList() {
+        arrList = new Object[DEFAULT_CAPACITY];
+    }
+
+    public void grow() {
+        T[] tempArrList = (T[]) arrList;
+        arrList = new Object[(int) (size * SCALE_FACTOR)];
+        System.arraycopy(tempArrList, FIRST_INDEX, arrList, FIRST_INDEX, size);
+    }
+
     @Override
     public void add(T value) {
-
+        if (arrList.length == size) {
+            grow();
+        }
+        arrList[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index:" + index + " out of size " + size);
+        }
+        if (arrList.length == size) {
+            grow();
```

Comment:
It will be better to move this 'if' conditional in resize() method to avoid code duplication. (duplicates in this method and in add(T value) as well).